lib/autoupdate/agent/setup.go

```diff
@@ -219,27 +241,63 @@ func (ns *Namespace) Teardown(ctx context.Context) error {
 	return nil
 }
 
-func (ns *Namespace) writeConfigFiles() error {
+func (ns *Namespace) writeConfigFiles(ctx context.Context) error {
 	var args string
 	if ns.name != "" {
 		args = " --install-suffix=" + ns.name
 	}
-	err := writeTemplate(
-		ns.updaterServiceFile, updateServiceTemplate,
-		struct{ UpdaterCommand string }{
+	err := writeTemplate(ns.updaterServiceFile, updateServiceTemplate,
+		struct {
+			UpdaterCommand string
+		}{
 			ns.updaterBinFile + args + " update",
 		},
 	)
 	if err != nil {
 		return trace.Wrap(err)
 	}
-	err = writeTemplate(
-		ns.updaterTimerFile, updateTimerTemplate,
-		struct{ TeleportService string }{filepath.Base(ns.serviceFile)},
+	teleportService := filepath.Base(ns.serviceFile)
+	err = writeTemplate(ns.updaterTimerFile, updateTimerTemplate,
+		struct {
+			TeleportService string
+		}{
+			teleportService,
+		},
```

Comment:
optional nit: this feels a bit awkward for the caller to create anonymous strucs, would it be more readable if we had the structs defined next to their templates? This way, when one edits the template, they also update the struct containing the template input.